### PR TITLE
volume-start: Allow requests without a body

### LINKED ISSUE #1065: failed to start a volume if request body is not present
curl -X POST http://192.168.122.253:24007/v1/volumes/mydistvol/start
{"errors":[{"code":1,"message":"EOF"}]}
Please refer start volume in  [Quick Start User Guide](https://github.com/gluster/glusterd2/blob/master/doc/quick-start-user-guide.md)

### LINKED ISSUE #1067: Volume start failing with REST-API
Volume start failing with REST-API, 

curl -X POST http://10.70.35.33:24007/v1/volumes/rep1/start
{"errors":[{"code":1,"message":"EOF"}]}

Logs:
time="2018-07-25 19:13:18.856318" level=info msg="10.70.35.33 - - [25/Jul/2018:19:13:18 +0530] \"POST /v1/volumes/rep1/start HTTP/1.1\" 400 40" reqid=ec27fc92-1970-4378-987d-2cd8c79f6fa6

### PR DESCRIPTION
**Pull request by prashanthpai**
Closes #1065 
Closes #1067 
Signed-off-by: Prashanth Pai <ppai@redhat.com>

### CODE REVIEW

**Comment by prashanthpai** [comment]
retest this please

**Comment by prashanthpai** [comment]
retest this please

File: glusterd2/commands/volumes/volume-start.go [hunk 1]
```diff
@@ -76,7 +77,8 @@ func volumeStartHandler(w http.ResponseWriter, r *http.Request) {
 	volname := mux.Vars(r)["volname"]
 	var req api.VolumeStartReq
 
-	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+	// request body is optional
+	if err := restutils.UnmarshalRequest(r, &req); err != nil && err != io.EOF {
```
Comment (aravindavk) [inline glusterd2/commands/volumes/volume-start.go]: just for clarification: if Unmarshal fails, will `req.Force` gets default value `false`?

Comment (prashanthpai) [reply glusterd2/commands/volumes/volume-start.go]: Yes

**Comment by prashanthpai** [comment]
retest this please

**Pull request merged**